core/src/main/scala/BinaryTree.scala

```diff
@@ -0,0 +1,250 @@
+/**
+ * Created by Nicolas A Perez (@anicolaspp) on 1/29/16.
+ */
+
+package dogs
+
+import Predef._
+import dogs.Order.{GT, EQ, LT}
+import scala.annotation.tailrec
+import scala.math
+
+
+
+abstract class BinaryTree[A] {
+
+  def value: Option[A]
+
+  val left: BinaryTree[A]
+  val right: BinaryTree[A]
+
+  private [dogs] val size: Int
+  private[dogs] val height: Int
+  private [dogs] def balanceFactor: Int
+  private[dogs] def update(): BinaryTree[A]
+
+  def isEmpty: Boolean
+
+  def toLst(): List[A] = this match {
+    case BTNil()                      =>  El[A]
+    case Branch(Some(a), l, r, _, _)  =>  l.toLst ::: (a :: r.toLst)
+  }
+
+  def inorder() = toLst()
+
+  def preorder(): List[A] = this match {
+    case BTNil()                      =>  El[A]
+    case Branch(Some(a), l, r, _, _)  =>  a :: l.preorder ::: r.preorder
+  }
+
+  def posorder(): List[A] = this match {
+    case BTNil()                      =>  El[A]
+    case Branch(Some(a), l, r, _, _)  =>  l.posorder() ::: r.posorder() ::: Nel(a, El[A])
+  }
+
+  def min(): Option[A] =  {
+    @tailrec def loop(sub: BinaryTree[A], x: Option[A]): Option[A] = sub match {
+      case BTNil()              =>  x
+      case Branch(a, l, _, _, _)      =>  loop(l, a)
+    }
+
+    this match {
+      case BTNil()                    => None()
+      case Branch(a, l, _, _, _)      =>  loop(l, a)
+    }
+  }
+
+  def max(): Option[A] = {
```

Comment:
no reason to have a nullary parameter group here. This is traditionally used to indicate that a method is performing side effects, but this method does not.

So this should be:

    def max: Option[A]

not 

    def max(): Option[A]